plugin.video.mediathekview/resources/lib/filmui.py

```diff
@@ -27,96 +39,146 @@ def Begin( self, showshows, showchannels ):
+			contextmenu.append( (
+				self.plugin.language( 30924 ),
+				'RunPlugin({})'.format(
+					self.plugin.build_url( {
+						'mode': "downloadep",
+						'id': self.id,
+						'quality': 1
+					} )
+				)
+			) )
+			if self.url_video_hd:
+				# Download HD TV episode
 				contextmenu.append( (
-					self.plugin.language( 30922 ),
-					'RunPlugin({})'.format( self.plugin.build_url( { 'mode': "download", 'id': self.id, 'quality': 0 } ) )
+					self.plugin.language( 30925 ),
+					'RunPlugin({})'.format(
+						self.plugin.build_url( {
+							'mode': "downloadep",
+							'id': self.id,
+							'quality': 2
+						} )
+					)
 				) )
-		# Add to queue
-		# TODO: Enable later
-#		contextmenu.append( (
-#			self.plugin.language( 30924 ),
-#			'RunPlugin({})'.format( self.plugin.build_url( { 'mode': "enqueue", 'id': self.id } ) )
-#		) )
-		li.addContextMenuItems( contextmenu )
+			listitem.addContextMenuItems( contextmenu )
+
+
+		if self.settings.autosub and self.url_sub:
+			videourl = self.plugin.build_url( {
+				'mode': "playwithsrt",
+				'id': self.id,
+				'external': False
+			} )
 
 		if totalItems is not None:
 			xbmcplugin.addDirectoryItem(
 				handle		= self.handle,
 				url			= videourl,
-				listitem	= li,
+				listitem	= listitem,
 				isFolder	= False,
 				totalItems	= totalItems
 			)
 		else:
 			xbmcplugin.addDirectoryItem(
 				handle		= self.handle,
 				url			= videourl,
-				listitem	= li,
+				listitem	= listitem,
 				isFolder	= False
 			)
 
 	def End( self ):
 		xbmcplugin.endOfDirectory( self.handle, cacheToDisc = False )
+
+	def get_list_item( self, alttitle, film = None ):
+		if film is None:
+			film = self
+		# get the best url
+		videourl = film.url_video_hd if ( film.url_video_hd != "" and self.plugin.settings.preferhd ) else film.url_video if film.url_video != "" else film.url_video_sd
+		videohds = " (HD)" if ( film.url_video_hd != "" and self.plugin.settings.preferhd ) else ""
+		# exit if no url supplied
+		if videourl == "":
+			return None
+
+		if alttitle is not None:
+			resultingtitle = alttitle
+		else:
+			if self.showshows:
+				resultingtitle = film.show + ': ' + film.title
+			else:
+				resultingtitle = film.title
+			if self.showchannels:
+				resultingtitle += ' [' + film.channel + ']'
+
+		info_labels = {
+			'title' : resultingtitle + videohds,
+			'sorttitle' : resultingtitle.lower(),
+			'tvshowtitle' : film.show,
+			'plot' : film.description
+		}
+
+		if film.size > 0:
+			info_labels['size'] = film.size * 1024 * 1024
+
+		if film.seconds > 0:
+			info_labels['duration'] = film.seconds
+
+		if film.aired is not None:
+			airedstring = '%s' % film.aired
+			if airedstring[:4] != '1970':
+				info_labels['date']		= airedstring[8:10] + '-' + airedstring[5:7] + '-' + airedstring[:4]
+				info_labels['aired']		= airedstring
+				info_labels['dateadded']	= airedstring
+
+		icon = 'special://home/addons/' + self.plugin.addon_id + '/resources/icons/' + film.channel.lower() + '-m.png'
```

Comment:
`xbmcaddon.Addon().getAddonInfo("icon")`

Reply:
This is not correct. According to [the doc](https://codedocs.xyz/AlwinEsch/kodi/group__python__xbmcaddon.html#ga65a28456461e069aeeddd13a8a4222fb) `xbmcaddon.Addon().getAddonInfo("icon")` would return the path to the addon icon as specified in the `addon.xml` file (`<icon>resources/icon.png</icon>`) but in case of channel ARD the icon would be `ADDONPATH/resources/icons/ard-m.png`.

Reply:
Disregard (though it was the addon icon), please use `os.path.join(xbmcaddon.Addon().getAddonInfo("path"), resources, icons, film.channel.lower() + '-m.png')` instead. Main reasoning is to use the python api instead of using the `special://` paths

Reply:
OK - will fix.